File: 프로그래머스/2/42883. 큰 수 만들기/큰 수 만들기.py
def solution(number, k):
    stack = []

    for num in list(map(int,list(number))):
        while(len(stack) > 0 and stack[-1] < num and k > 0):
            k-=1
            stack.pop()
        stack.append(num)

    stack = "".join(list(map(str,list(stack))))
    return stack[:-k] if k > 0 else stack

File: 프로그래머스/2/42883. 큰 수 만들기/큰 수 만들기.java
import java.util.*;
import java.util.stream.*;
class Solution {
    public String solution(String number, int k) {
        String answer = "";
        Stack<Integer> stack = new Stack<>();
        for(int i = 0; i<number.length(); i++){
            Integer num = Integer.parseInt(number.substring(i, i+1));
            while(!stack.isEmpty() && k>0 && stack.peek() < num){
                k--;
                stack.pop();
            }
            stack.push(num);
        }
        for(Integer s : stack){
            answer += String.valueOf(s);
        }
        return answer.substring(0, answer.length()-k);
    }
}